queue/retry.go

```diff
@@ -11,12 +11,36 @@ import (
 	"github.com/mongodb/grip/message"
 	"github.com/mongodb/grip/recovery"
 	"github.com/pkg/errors"
+	"go.mongodb.org/mongo-driver/mongo"
 )
 
+// statsRetryHandler is the same as an amboy.RetryHandler but allows it to
+// provide additional runtime statistics on its jobs. This is primarily for
+// queue-internal tracking and testing purposes.
+type statsRetryHandler interface {
+	amboy.RetryHandler
+	// stats returns information about the job statistics for an
+	// amboy.RetryHandler. Results are reset with each runtime execution.
+	stats() retryHandlerStats
+}
+
+// retryHandlerStats captures runtime statistics on an amboy.RetryHandler.
+type retryHandlerStats struct {
+	pending   int
+	completed int
+	retried   int
```

Comment:
Can we call this something else to make this more obvious that it is an aggregated count of errored completed retry jobs? Maybe it is just me, but I think at first glance `retried` isn't obvious. I didn't really get what the stat was telling us until I read L210.

Reply:
These are temporary stats until [EVG-14111](https://jira.mongodb.org/browse/EVG-14111) is done, so it doesn't seem worth it.

Reply:
SGTM